docs/code_ref/coordinates.rst

```diff
@@ -114,19 +113,72 @@ Both the heliographic frames use latitude, longitude and radius which are access
+
 
 Observer Location Information
 -----------------------------
 
-Both ``Helioprojective`` and ``Heliocentric`` frames are defined by the location of the observer. For example in ``Helioprojective`` the observer is at the origin of the coordinate system. This information is encoded in the ``Helioprojecitve`` and ``Heliocentric`` frames as the attributes `L0`, `B0` and `D0` which are heliographic longitude, latitude and distance from the center of the Sun. These attributes are automatically populated from meta data when coordinate frames are created using map.
-
-It is possible to convert from a ``Helioprojective`` frame with one observer location to another ``Helioprojecitve`` frame with a different observer location, by converting through ``Heliographic``, this does involve making an assumption of the radius of the Sun to calculate the position on the solar sphere. The conversion can be performed as follows::
+Both `~sunpy.coordinates.frames.Helioprojective` and
+`~sunpy.coordinates.frames.Heliocentric` frames are defined by the location of
+the observer. For example in `~sunpy.coordinates.frames.Helioprojective` the
+observer is at the origin of the coordinate system. This information is encoded
+in the `~sunpy.coordinates.frames.Helioprojective` and
+`~sunpy.coordinates.frames.Heliocentric` frames as the ``observer`` attribute,
+which is itself an instance of the
+`~sunpy.coordinates.frames.HeliographicStonyhurst` frame. The default observer
+location is set to the position of the Earth (using
+`~sunpy.coordinates.ephemeris.get_body_heliographic_stonyhurst`) as long as the
+``obstime`` attribute is specified. If the ``obstime`` attribute is not set the
+observer defaults to ``(0°, 0°, 1 AU)`` i.e. the mean position of the Earth. The
+location of the observer is automatically populated from meta data when
+coordinate frames are created using map.
+
+It is possible to convert from a `~sunpy.coordinates.frames.Helioprojective`
+frame with one observer location to another
+`~sunpy.coordinates.frames.Helioprojective` frame with a different observer
+location, by converting through `~sunpy.coordinates.frames.Heliographic`, this
+does involve making an assumption of the radius of the Sun to calculate the
+position on the solar sphere. The conversion can be performed as follows::
 
   # Input coordinate
-  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, frame=frames.Helioprojective)
-  # Define the location of the new observer as a Helioprojective frame
-  >>> hpc_out = sunpy.coordinates.Helioprojective(L0=10*u.deg)
-  # Perform the conversion
+  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, observer="earth", obstime="2017-07-26", frame=frames.Helioprojective)
+  # Define a new Helioprojective frame with a different observer.
+  >>> hpc_out = sunpy.coordinates.Helioprojective(observer="venus", obstime="2017-07-26")
+  # Perform the transformation from one to the other.
   >>> hpc2 = hpc1.transform_to(hpc_out)
```

Comment:
My guess is that HELIO's `hci` is a coordinate system fixed in space (e.g., to the equinox, as @dpshelio already suggested) rather than rotating with the Earth (HGS) or with Carrington's defined rotation (HGC).  This code returns Carrington longitudes to within two arcminutes of the *Astronomical Almanac*, so HELIO can't be returning those.